module/merger.go

```diff
@@ -8,6 +8,6 @@ import (
 // in a cryptographically unaware way (agnostic of the byte structure of the
 // signatures).
 type Merger interface {
-	Join(sigs ...crypto.Signature) ([]byte, error)
-	Split(combined []byte) ([]crypto.Signature, error)
+	Join(sig1, sig2 crypto.Signature) []byte
+	Split(combined []byte) (crypto.Signature, crypto.Signature, error)
```

Comment:
Isn't the internal notion of expected length related to the specific implementation rather than the interface?  
I like adding an error output for the method `Join`, but maybe we can keep it generic in the interface, and detail the error reasons in the implementation module/signature/combine.go ?

Reply:
The problem is that higher level-code only sees the interface. Hence, if I am working with the interface, I still have to handle the errors; and I need to decide which errors are fatal and which errors are expected as part of normal operations. 

What you are suggesting is: 
* Use the interface but write the code which is using the interface with knowledge about the specific implementation that we _expect_ to be hidden behind the implementation. What is the point of the interface then? 

My view is:
* everything that I need to know, should be documented in the interface. 
* If I have to handle specific sentinel errors as part of normal operations, those errors must be documented in the interface (because all implementations need to have the same convention). 

In our particular case it means:
* Split can through an error. 
* Is this error universally fatal? No! (because wrong signatures generated by a malicious node have to be gracefully handled by honest nodes)
* Hence, I have to check for _specific sentinel_ errors in the code which uses the `Merger` interface. 
* Hence, these errors cannot be considered an internal implementation detail and need to be documented in the interface.

Reply:
Just for curiosity:
I think for all protocol-related signatures, we have a notion of "expected length", right? Hence, is there any harm in reflecting this in the interface? 

I think we don't need to make interfaces more general than they have to be.

Reply:
I understand your point about differentiating sentinel errors from other errors. Actually I'm not a big fun of writing interfaces with a single implementation, this just results in 2 type names and 2 similar interfaces, which is just confusing in my opinion 😅 

I don't have a strong opinion about where this specific comments should live. I only suggested not to add it to the interface because the previous implementation didn't use any notion of internal or expected length and still implemented the same `Merger` interface. A joined signature in the previous implementation used to be `len(sig1) || sig1 || len(sig2) || sig2` .
So for instance I was thinking of still keeping the sentinel error `verification.ErrInvalidFormat` and mention the wrong format case, but without mentioning the internal expected length, as some implementations may not have one.

Anyway, I'll copy the comments to the interface as you suggested 👌 Thank you for clarifying your point.

Reply:
We could also remove the interface and just work with concrete `Merger` instances (?)

Reply:
I can plan that for the wider crypto V2 refactor.